doc/policy_based_metering/PBM-HLD.md

```diff
@@ -0,0 +1,352 @@
+* packet_action_mirror = True/false                   ; available to mirror acl table type).
+* packet_action_do_not_nat = True/false       
+* packet_action_policer = True/false          
+
+redirect_action = 1*255CHAR                           ; redirect parameter
+                                                      ; This parameter defines a destination for redirected packets
+                                                      ; it could be:
+                                                      : name of physical port. Example: "Ethernet10"
+                                                      : name of LAG port       Example: "PortChannel5"
+                                                      
+mirror_action = 1*255VCHAR                            ; refer to the mirror session
+                                                      ; (only available to mirror acl
+                                                      ; table type)
+                                                      
+* policer_name = 1*255VCHAR                           ; refer to the policer object name
+           
+ether_type    = h16                                   ; Ethernet type field
+           
+ip_type       = ip_types                              ; options of the l2_protocol_type
+                                                      ; field. Only v4 is support for
+                                                      ; this stage.
+           
+ip_protocol   = h8                                    ; options of the l3_protocol_type field
+           
+src_ip        = ipv4_prefix                           ; options of the source ipv4
+                                                      ; address (and mask) field
+           
+dst_ip        = ipv4_prefix                           ; options of the destination ipv4
+                                                      ; address (and mask) field
+           
+l4_src_port   = port_num                              ; source L4 port or the
+l4_dst_port   = port_num                              ; destination L4 port
+.
+.
+.
+```
+
+##### Example for JSON config file with the new fields:
+```JSON
+{
+    "POLICER_TABLE|M_POLICER_7": {
+        "meter_type": "packets",
+        "mode": "tr_tcm",
+        "color": "aware",
+        "cir": "5000",
+        "cbs": "5000",
+        "green_packet_action": "forward",
+        "red_packet_action": "drop"
+    },
+    
+    "ACL_TABLE|Table_ID_#1": {
+        "policy_desc": "Permit some traffic",
+        "type": "L3",
+        "ports": [
+            "Ethernet2",
+            "Ethernet4",
+            "Ethernet7"
+        ],
+      * "ext_actions_list": "POLICER",
+        "OP": "SET"
+    },
+    
+    "ACL_RULE|Table_ID_#1|Rule_ID_#1": {
+        "priority": "70",
+        "packet_action": "FORWARD",
+      * "packet_action_policer": "True",
+      * "policer_name": M_POLICER_7,   
+        "IP_PROTOCOL": "TCP",
```

Comment:
if we list specific policer fields here - why is ext_actions_list needed?

Reply:
When a new ACL is created, an action list that defines the actions that the ACL rules can be used is sent to SAI. The action of the rules can only be a sub-set of the ext_actions_list+default_actions.